streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -184,7 +184,8 @@ private synchronized void setStateWhenNotInPendingShutdown(final State newState)
 
     protected final StreamsConfig config;
     protected final TopologyBuilder builder;
-    protected final Producer<byte[], byte[]> producer;
+    protected Producer<byte[], byte[]> threadProducer = null;
```

Comment:
Yes. I like to add it explicitlye -- personal taste. Can remove it if it bothers you :)